### DISCUSSION TITLE
filter vs. map to update a useState for rendering (delete button)

### DISCUSSION THREAD

**Discussion by DanielGennat** [neuefische/questions#240] [Web] [answered]
Hallo, ein kurze Verständnisfrage.
Ich habe gerade einen delete Button geschrieben, der das letzte Objekt aus meinem useState löscht, mit dessen Objekten ein Formular gerendert wird.
Wenn ich die handleRemove analog zur handleAdd schreibe, bekomme ich beim Rendern eine Fehlermeldung, dass die Values vom gelöschten Object nicht gelesen werden können.
```
export function handleRemove(newTimerChain, setNewTimerChain) {
  const lastId = newTimerChain.length;
  setNewTimerChain(
    newTimerChain.map((timer) => {
      if (timer.id !== lastId) {
        return timer;
      }
    })
  );
}
```
Die setNewTimerChain updatet die newTimerChain also anscheinend nicht 
Wenn ich filter verwende funktioniert es:
```
export function handleRemove(newTimerChain, setNewTimerChain) {
  const lastId = newTimerChain.length;
  const newChain = newTimerChain.filter((timer) => timer.id !== lastId);
  setNewTimerChain(newChain);
}
```
Warum funktioniert es nicht mit map? :)

**Comment by NikRoe** [comment, marked as answer]
Hi Daniel,

using `.map()` you can modify the elements of the original array and save them in a second array. Mapping over an array creates a new array of the same length as the original array with potentially modified elements. [More info.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map)

Using `.filter()` you can remove elements from an array. Filtering over an array creates a new array that is potentially shorter than the original array. The elements within the array are not modified. [More info.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter)

--> If you want to remove an element, you should use `.filter()`. If you wish to modify elements, you should use `.map()`.

Hope this helps :) 

-Niklas

**Reply by DanielGennat** [reply]
It does! Thanks! :)
